DEVELOPMENT.md

```diff
@@ -4,6 +4,99 @@
 
 * See [README.md](./README.md).
 
+## Local clusters
+
+When working on compliantkubernetes-apps it is possible to work with local clusters running on [kind](https://kind.sigs.k8s.io/).
+
+This requires that `kind` is installed and that either `podman` or `docker` is available and setup to work with `kind`.
+
+### Setup
+
+> [!warning]
+> Ensure your `KUBECONFIG` variable is not pointing to a config that you don't want to get edited by `kind`.
+
+> [!important]
+> Ensure your `inotify` limits are high enough else [pods might fail with the "too many open files" or similar](https://kind.sigs.k8s.io/docs/user/known-issues/#pod-errors-due-to-too-many-open-files)
+>
+> ```sh
+> sysctl fs.inotify.max_user_instances # at least 1024 recommended
+> sysctl fs.inotify.max_user_watches   # at least 102400 recommended
+> ```
+
+> [!important]
+> With rootless `podman` or `docker` ensure your unprivileged port limit starts at `53`.
+>
+> ```sh
+> sysctl net.ipv4.ip_unprivileged_port_start # expected at 53
+> ```
+
+```sh
+# with CK8S_CONFIG_PATH and CK8S_PGP_FP set
+./scripts/local-cluster.sh config <name> <apps-flavor> <domain>
+./scripts/local-cluster.sh create <name> <kind-config|local-cluster-flavor> # use list flavors to see built-in configs
+```
+
+This will configure apps with `ck8sCloudProvider: none` and `ck8sFlavor: <apps-flavor>` and set some default values to run on local clusters.
+By default it will use `calico` for networking, `local-path-provisioner` for block storage, and `minio` for object storage.
+
+The configuration contains some `set-me`'s that must be configured manually.
+
+### Deploy
+
+> [!important]
+> Setting up ingresses properly requires some additional steps documented later in this section.
+
+Manage apps by using `helmfile` directly and with needs it will pull in all required releases:
+
+```sh
+# for service cluster
+helmfile -e service_cluster <operation> --selector app=<application> --include-transitive-needs
+# for workload cluster
+helmfile -e workload_cluster <operation> --selector app=<application> --include-transitive-needs
+```
+
+Use `helmfile -e <service|workload>_cluster list` to list all releases and to view their labels.
+By default all releases have `name=<release-name>` and `chart=<chart-name>` as predefined labels.
+
+Enabling ingress and resolve requires a special setup for `ingress-nginx` and `node-local-dns` respectively.
+Both will by default be port-mapped on the local address `127.0.64.43` and once both are running you need to change your computer's network settings to use this address as the DNS resolver, then you can access service endpoints using the configured domain.
+
+Both are already pre-configured by `scripts/local-cluster.sh` and can be deployed by using:
+
+```sh
+helmfile -e <service|workload>_cluster -lapp=ingress-nginx -lapp=node-local-dns apply --include-transitive-needs
```

Comment:
This fails for me due to missing either the `monitoring` (for both) and `hnc-system` (only wc) namespaces.

Reply:
Did you run bootstrap as I noted on Slack? This is a known issue as it is in a different PR.

I forgot to write it here.

Reply:
Right, sorry. That had scrolled out of view 😄